sklearn/preprocessing/value_dropper.py

```diff
@@ -0,0 +1,216 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+
+from ..utils import check_X_y
+from ..utils import check_random_state
+
+__all__ = ["drop_values"]
+
+
+def drop_values(X, y=None, missing_mask=None,
+                missing_values="NaN",
+                missing_fraction=0.1,
+                label_correlation=1.0,
+                n_labels=1,
+                labels=None,
+                missing_mask_only=False,
+                return_labels=False,
+                return_missing_mask=False,
+                copy=False,
+                random_state=None):
+    """Drop values based on a preset strategy.
+
+    Attributes
+    ----------
+
+    X : ndarray like of shape (n_features, n_samples)
+
+        Data, in which the values must be dropped and set to
+        missing_values.
+
+    y : ndarray like of shape (n_samples,), optional
+        Target, in a supervised classification task.
+
+    missing_mask : bool ndarray shape (n_features, n_samples), optional
+        This is used to either denote the missing values that already
+        exist in the data or simply to specify a missing mask to modify
+        inplace.
+
+    missing_values : {"NaN" (or np.nan) | int}, default "NaN"
+        The missing value to use
+
+    label_correlation : float, default 1.0
+        1 (MNAR) - Randomly choose n_targets (or take from given
```

Comment:
Not sure what `n_targets` means. Surely you're not referring to a multi-target problem; your `y` is of shape `(n_samples,)`. When you say targets, do you mean classes? Do you mean the parameter `n_labels`??